### ISSUE TITLE
'CombinedExpression' object is not iterable when using an admin filter on Django 1.8

### ISSUE THREAD

**Issue by ALenfant** [disqus/django-bitfield#64]
Hi,

I'm having a problem when using `BitFieldListFilter` on Django 1.8

Selecting any item in the filter raises the following error:

```
Traceback:
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/contrib/admin/options.py" in wrapper
  616.                 return self.admin_site.admin_view(view)(*args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  110.                     response = view_func(request, *args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/views/decorators/cache.py" in _wrapped_view_func
  57.         response = view_func(request, *args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/contrib/admin/sites.py" in inner
  233.             return view(request, *args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapper
  34.             return bound_func(*args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/utils/decorators.py" in _wrapped_view
  110.                     response = view_func(request, *args, **kwargs)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/utils/decorators.py" in bound_func
  30.                 return func.__get__(self, type(self))(*args2, **kwargs2)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/contrib/admin/options.py" in changelist_view
  1548.                 self.list_max_show_all, self.list_editable, self)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/contrib/admin/views/main.py" in __init__
  82.         self.get_results(request)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/contrib/admin/views/main.py" in get_results
  177.         result_count = paginator.count
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/core/paginator.py" in _get_count
  77.                 self._count = len(self.object_list)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/query.py" in __len__
  144.         self._fetch_all()
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/query.py" in _fetch_all
  965.             self._result_cache = list(self.iterator())
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/query.py" in iterator
  238.         results = compiler.execute_sql()
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in execute_sql
  829.             sql, params = self.as_sql()
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in as_sql
  387.             where, w_params = self.compile(self.query.where)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in compile
  357.             sql, params = node.as_sql(self, self.connection)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/sql/where.py" in as_sql
  104.                     sql, params = compiler.compile(child)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/sql/compiler.py" in compile
  357.             sql, params = node.as_sql(self, self.connection)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/django/db/models/lookups.py" in as_sql
  210.         lhs_sql, params = self.process_lhs(compiler, connection)
File "/Users/antonin/Documents/Projects/venv/lib/python3.4/site-packages/bitfield/query.py" in process_lhs
  38.             params.extend(self.get_db_prep_lookup(self.rhs, connection)[1])

Exception Type: TypeError at /admin/core/item/
Exception Value: 'CombinedExpression' object is not iterable
```

It looks like this should be fixed for Django 1.8

**Comment by ALenfant** [comment]
After trying to find a solution myself, I looked through the forks and found this bit of code that fixed my issue: https://github.com/iamahuman/django-bitfield/commit/8b8aade8d617906d80f0c188fb7647b748eace5f

I think it should be integrated if it doesn't cause any side-effects (I couldn't see any).

**Comment by giorgosp** [comment]
Is this going to be merged soon?

**Comment by ideallical** [comment]
+1

**Comment by DylannCordel** [comment]
:+1: Same error here with Django 1.8

**Comment by DylannCordel** [comment]
this problem is still there with the last released version of django-bitfield (1.9.1)

**Comment by Datamance** [comment]
This is _still_ a problem with django-bitfield==1.9.3 - I get `django.db.utils.ProgrammingError: can't adapt type 'CombinedExpression'` when trying to filter in admin.

**Comment by timabbott** [comment]
Can you test whether this is fixed in the 1.9.5 release?

I don't have time to debug myself, but I'd happily merge a patch for this and do a new release.

**Comment by timabbott** [comment]
I believe this is fixed in #103.
